sdk-extensions/incubator/src/main/java/io/opentelemetry/sdk/extension/incubator/fileconfig/LogRecordExporterFactory.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.extension.incubator.fileconfig;
+
+import static java.util.stream.Collectors.joining;
+
+import io.opentelemetry.sdk.autoconfigure.internal.NamedSpiManager;
+import io.opentelemetry.sdk.autoconfigure.internal.SpiHelper;
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigurationException;
+import io.opentelemetry.sdk.autoconfigure.spi.internal.DefaultConfigProperties;
+import io.opentelemetry.sdk.autoconfigure.spi.logs.ConfigurableLogRecordExporterProvider;
+import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.Otlp;
+import io.opentelemetry.sdk.logs.export.LogRecordExporter;
+import java.io.Closeable;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+final class LogRecordExporterFactory
+    implements Factory<
+        io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.LogRecordExporter,
+        LogRecordExporter> {
+
+  private static final LogRecordExporterFactory INSTANCE = new LogRecordExporterFactory();
+
+  private LogRecordExporterFactory() {}
+
+  static LogRecordExporterFactory getInstance() {
+    return INSTANCE;
+  }
+
+  @Override
+  public LogRecordExporter create(
+      @Nullable
+          io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.LogRecordExporter
+              model,
+      SpiHelper spiHelper,
+      List<Closeable> closeables) {
+    if (model == null) {
+      return LogRecordExporter.composite();
+    }
+
+    if (model.getOtlp() != null) {
+      Otlp otlp = model.getOtlp();
+
+      Map<String, String> properties = new HashMap<>();
+      if (otlp.getProtocol() != null) {
+        properties.put("otel.exporter.otlp.logs.protocol", otlp.getProtocol());
```

Comment:
We intentionally want to avoid any dependencies on the exporter modules so that the logic for configuring an exporter from `ConfigProperties` can be a distributed problem and encapsulated in the respective module. 

This is a bit of a smell, but we're essentially converting between the file configuration schema and the environment variable schema, and I think its acceptable to have that translation live in a central place, even if the code interpreting the config is distributed.

Reply:
👍 Consider adding a comment to this effect in the class.